jaeger-core/src/test/java/io/jaegertracing/ConfigurationTest.java

```diff
@@ -328,6 +333,19 @@ public void testNoServiceName() {
     new Configuration(null);
   }
 
+  @Test(expected = RuntimeException.class)
+  public void testEmptyServiceName() {
+    new Configuration("");
```

Comment:
I actually thought about having the `null` overriding a valid existing name:

```java
new Configuration("Test").withServiceName(null);
```

This way, you'd be documenting the expected behavior for this scenario, as it might be ambiguous. This is not required, so, I'm giving my +1 to this PR.

Reply:
The check is also in tracer constructor, but in this case we wanted to fail before tracer-init time